### PR TITLE
Add ability to check if layout contains region

### LINKED ISSUE #2594: RegionManager#hasRegion(regionName)
I was surprised to find that this doesn't exist!

`return Boolean(this.getRegion(regionName));`

Also, it goes without saying, but it should be added to the LayoutView prototype.

### PR DESCRIPTION
**Pull request by jisaacks**
fixes #2594

### CODE REVIEW

**Comment by samccone** [comment]
@jisaacks Hi thanks for the PR. Can you describe the usecase here, and why you feel like it belongs in core?

**Comment by jisaacks** [comment]
@samccone I just assumed that since one of this project's owners (@jmeas) opened an issue asking for it, it was wanted. If it is not, feel free to close.

**Comment by samccone** [comment]
Sure that is a fair assumption, I was wondering why do you think it is a good idea?

---
@jmeas @ahumphreys87 what are your motivations for this feature

**Comment by jisaacks** [comment]
@samccone 

>  I was wondering why do you think it is a good idea?

It seems to me that it naturally fits in the API along with the existing, `getRegion`, `addRegion`, `removeRegion`. In my use of Marrionette, my layoutViews have generally been pretty static, so I would not make arguments to persuade anyone to include this feature other than completing the API.

**Comment by jmeas** [comment]
@samccone, the `showChildView` method I created in [this issue](https://github.com/marionettejs/backbone.marionette/issues/2593) is the motivation here. The two features go hand-in-hand.

**Comment by samccone** [comment]
:+1: sgtm Just wanted to have a little bit of understanding with what you were  thinking :clap:

File: src/layout-view.js [hunk 1]
```diff
@@ -88,6 +88,11 @@ Marionette.LayoutView = Marionette.ItemView.extend({
     return this.regionManager.removeRegion(name);
   },
 
+  // Checks to see if layout contains region
+  hasRegion: function(name) {
```
Comment (samccone) [inline src/layout-view.js]: @jisaacks Mind adding docs to the layout readme and the layout yaml

**Comment by jisaacks** [comment]
@samccone let me know if there is anything else.

**Comment by jasonLaster** [comment]
looks good to me, but this should go into `next` and not `master`

**Comment by jisaacks** [comment]
I'll make a new PR

**Comment by jisaacks** [comment]
Actually I will have to look at this later @jasonLaster I thought I could just checkout the **next** branch and cherry pick my commits over. Doing so causes a strange merge conflict with irrelevant files, and it looks like `src/layout-view.js` doesn't event exist in the **next** branch. I don't have time to figure that out at the moment, maybe later tonight I can take another look.

**Comment by ahumphreys87** [comment]
ah! @jisaacks yeah in next `LayoutView` and `ItemView` have been merged into `View`, it _should_ be safe to lift your changes into there.

**Comment by jisaacks** [comment]
@jasonLaster @ahumphreys87 PR against next [here](https://github.com/marionettejs/backbone.marionette/pull/2627)

**Comment by jisaacks** [comment]
Also, what is the purpose of the **next** branch? I assumed v3 but it's package.json says 2.4.1

**Comment by jmeas** [comment]
manifest updates are the last thing we do

**Comment by ahumphreys87** [comment]
fix merged with #2627

**Pull request closed without merging**